design/specs/specs-null-values.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Null Value Specs
+description: 
+published: true
+date: 2021-11-25T09:00:00.201Z
+tags: 
+editor: markdown
+dateCreated: 2021-11-25T09:00:00.201Z
+---
+
+## Context
+
+NULL values are used to indicate that a value does not exist in a database. It is important to make it clear to users that NULL is not the same as blank or zero. Ideally, we want users to think of NULL values as placeholders for unknown or missing values that are applicable.
+
+## Prototype
+
+### Link
+
+[Prototype](https://mathesar-prototype.netlify.app/?activeTable=2)
+
+### Videos
+
+- Scenario 1: [Video](https://www.loom.com/share/527224982d534a2db082359360147ad4)
+- Scenario 2: [Video](https://www.loom.com/share/7de3c90af3d74ac195397c643f318867)
+- Scenario 3: [Video](https://www.loom.com/share/7ea4ca9fe3e14acdbffac5ea8e77e760)
+
+## User Experience
+
+### Considerations for handling NULL Values
+
+When presented in the form of tabular data, NULL values should be visually distinct so that the user understands that they are not the same as other string types. The style should attempt to communicate the absence of a value, similar to how placeholders are used in text input fields.
+As for the process of transforming values to and from NULL, it will be important to have a consistent way of doing this across different data types.
+
+## Scenarios
+
+### Scenario 1: User identifies a NULL Value
+
+#### Scenario 1a: The user identifies a NULL value in a table
+
+#### Steps for 1a
```

Comment:
Just a style issue, I would remove these "Steps" headers because they don't add any extra information and it's odd to have two headers at the same level.

Reply:
@kgodey It was mostly for consistency if I had other points that were not steps. So for some, I'd have steps and not for others.